components/calendar/src/julian.rs

```diff
@@ -0,0 +1,380 @@
+        )
+    }
+
+    /// The calendar-specific year represented by `date`
+    fn year(&self, date: &Self::DateInner) -> types::Year {
+        date.0.year.into()
+    }
+
+    /// The calendar-specific month represented by `date`
+    fn month(&self, date: &Self::DateInner) -> types::Month {
+        date.0.month.into()
+    }
+
+    /// The calendar-specific day-of-month represented by `date`
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day.into()
+    }
+
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = IsoYear(date.0.year.0 - 1);
+        let next_year = IsoYear(date.0.year.0 + 1);
+        types::DayOfYearInfo {
+            day_of_year: Julian::day_of_year(date.0),
+            days_in_year: Julian::days_in_year(date.0.year),
+            prev_year: prev_year.into(),
+            days_in_prev_year: Julian::days_in_year(prev_year),
+            next_year: next_year.into(),
+        }
+    }
+
+    fn debug_name() -> &'static str {
+        "Julian"
+    }
+}
+
+impl Julian {
+    /// Construct a new Julian Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    fn is_leap_year(year: IsoYear) -> bool {
+        year.0 % 4 == 0
+    }
+
+    fn days_in_year(year: IsoYear) -> u32 {
+        if Self::is_leap_year(year) {
+            366
+        } else {
+            365
+        }
+    }
+
+    fn day_of_year(date: IsoDateInner) -> u32 {
+        let month_offset = [0, 1, -1, 0, 0, 1, 1, 2, 3, 3, 4, 4];
+        let mut offset = month_offset[u8::from(date.month) as usize - 1];
+        if Self::is_leap_year(date.year) && u8::from(date.month) > 2 {
+            offset += 1;
+        }
+        let prev_month_days = (30 * (u8::from(date.month) as i32 - 1) + offset) as u32;
+        prev_month_days + u8::from(date.day) as u32
+    }
+
+    fn days_in_month(year: IsoYear, month: IsoMonth) -> u8 {
+        match u8::from(month) {
+            4 | 6 | 9 | 11 => 30,
+            2 if Self::is_leap_year(year) => 29,
+            2 => 28,
+            _ => 31,
+        }
+    }
+
+    fn fixed_from_julian(date: IsoDateInner) -> i32 {
+        let mut fixed: i32 = -1 + 365 * (date.year.0 - 1) + (date.year.0 - 1) / 4;
+        fixed = fixed + (367 * (u8::from(date.month) as i32) - 362);
+        fixed = fixed
+            + if u8::from(date.month) <= 2 {
+                0
+            } else if Self::is_leap_year(date.year) {
+                -1
+            } else {
+                2
+            };
+
+        return fixed + u8::from(date.day) as i32;
+    }
+
+    fn fixed_from_julian_integers(year: i32, month: i32, day: i32) -> i32 {
+        Self::fixed_from_julian(
+            *Date::new_iso_date_from_integers(year, month as u8, day as u8)
+                .unwrap()
+                .inner(),
+        )
+    }
+
+    fn julian_from_fixed(date: i32) -> JulianDateInner {
+        let approx = 1464 + (4 * date) / 1461;
```

Comment:
issue: this is supposed to be `4 * date - julian_epoch`, where `julian_epoch` is the date of Julian Jan 1, 1, which is `-1`